reconcile/jira_watcher.py

```diff
@@ -0,0 +1,70 @@
+import os
+import json
+import logging
+
+import utils.gql as gql
+
+from utils.jira_client import JiraClient
+
+
+QUERY = """
+{
+  jira_boards: jira_boards_v1 {
+    path
+    name
+    serverUrl
+    token {
+      path
+    }
+  }
+}
+"""
+
+QONTRACT_INTEGRATION = 'jira-watcher'
+
+
+def get_project_file_path(io_dir, project):
+    dir_path = os.path.join(io_dir, QONTRACT_INTEGRATION)
+    if not os.path.exists(dir_path):
+        os.makedirs(dir_path)
+    return os.path.join(dir_path, project + '.json')
+
+
+def run(dry_run=False, io_dir='throughput/'):
+    gqlapi = gql.get_api()
+    jira_boards = gqlapi.query(QUERY)['jira_boards']
+    for jira_board in jira_boards:
+        jira = JiraClient(jira_board)
+        issues = jira.get_issues()
+        current_state = {issue.key: issue.fields.status.name
+                         for issue in issues}
+        project_file_path = get_project_file_path(io_dir, jira.project)
+
+        try:
+            with open(project_file_path, 'r') as f:
+                previous_state = json.load(f)
```

Comment:
tbh this struck me a bit odd at first, but after thinking about this for 5 minutes I agree this is the best solution, I mean keeping the previous state in the file, and overwriting it each cycle. My first impulse would have been to iterate through all issues and report if any changes have been in the last X minutes, where X minutes is the cadence of the job periodicity. But this makes much more sense.

awesome work @maorfr

Reply:
thanks for the cool feedback!